core-rust/state-manager/src/state_computer.rs

```diff
@@ -1210,12 +1271,55 @@ where
         self.committed_transactions_metrics
             .update(transactions_metrics_data);
 
+        let mut locked_protocol_state = self.protocol_state.write();
+        *locked_protocol_state = new_protocol_state;
+        drop(locked_protocol_state);
+
         Ok(CommitSummary {
             validator_round_counters: round_counters,
             num_user_transactions,
         })
     }
 
+    pub fn apply_protocol_update(
+        &self,
+        protocol_updater: &dyn ProtocolUpdater,
+    ) -> ProtocolUpdateResult {
```

Comment:
Minor: It feels a little weird that we need to pass in an Updater into a method of StateComputer. Perhaps better would be to only pass in the required state of StateComputer to Updater for updates?